src/bridge.js

```diff
@@ -374,8 +375,11 @@ Bridge.prototype.run = function(port, config, appServiceInstance, hostname) {
         this._metrics.addAppServicePath(this);
     }
 
-    this.appService.listen(port, hostname);
-    return this.loadDatabases();
+    // We MUST return a Bluebird-Promise instead of a Promise.
+    // promise.done() is used by many tests in this repo.
+    return this.loadDatabases().then(async() => {
```

Comment:
Stupid question, why don't we use await here?

Reply:
Great question.
`loadDatabase` must be executed immediately to initialize the stores needed by some promises above. These promises will run as soon as `run` is done. The only other way is to move `loadDatabase` above the usage of `Promise.cast()`&`this._onAliasQuery()` and await it there.

`loadDatabase` returns a Bluebird-Promise, which is exactly what I need to return to make the tests happy. I just chain `.listen` in a native promise at the end, but the return value will stay a Bluebird-Promise.

Reply:
Fair enough, what a mess though :S